### DISCUSSION TITLE
Update the Hugo binary

### DISCUSSION THREAD

**Discussion by salim-b** [gethyas/hyas#418] [Q&A] [answered]
hyas uses [hugo-installer](https://github.com/dominique-mueller/hugo-installer/) to fetch the Hugo binary for the proper platform and architecture. The Hugo *version* to fetch is configured in hyas' `package.json` under key `otherDependencies.hugo`:

https://github.com/gethyas/hyas/blob/e858a9ec337f4ee3dacd67d1e762fd05a6fbb340/package.json#L58-L60

In the Doks child theme, we could simply set that key in our main project's `package.json` to define the Hugo version to be used since the default `postinstall` command there was `hugo-installer --version otherDependencies.hugo --extended --destination node_modules/.bin/hugo`.

In the new approach where Doks is a theme for Hyas, it seems the Hugo version is not supposed to be altered by the user anymore since the default `postinstall` command now just [copies over the Hugo version defined by the hyas package](https://github.com/gethyas/create-hyas/blob/main/template-doks/package.json#L25). Users who want to use a newer Hugo version have to wait for the hyas npm package to include that Hugo version. I think this is overly restrictive.

Any plans on changing this and allowing the user to define the Hugo version by default?

**Comment by h-enk** [comment]
Thanks for your input! Get back to you next Wednesday (when I'm back home).

**Comment by h-enk** [comment, marked as answer]
> Any plans on changing this and allowing the user to define the Hugo version by default?

No, this is by (new) design — to guarantee you're always using a Hugo version that works with (id est is tested for) a specific Hyas version. I understand that this could be restrictive for some users. However, I'm planning to set up automated tests (and automated publishing) to keep things fairly easily up-to-date.

**Reply by salim-b** [reply]
Ok, I see. Thanks for the clarification.

Just for reference: Users who want to take control over the Hugo version in use can simply replace the `postinstall` command and add `otherDependencies.hugo` with the desired Hugo version in their site's main `package.json`:

```json
{
  "scripts": {
    "postinstall": "shx rm -rf node_modules/.bin/hugo && hugo-installer --version otherDependencies.hugo --extended --destination node_modules/.bin/hugo"
  },
  "otherDependencies": {
    "hugo": "0.117.0"
  }
}
```

And execute `npm run postinstall` after changing the value of  `otherDependencies.hugo`.

**Reply by h-enk** [reply]
Good addition — I will add that to the documentation (as an option)
